app/workers/push_notifications/cleanup_worker.rb

```diff
@@ -0,0 +1,18 @@
+module PushNotifications
+  class CleanupWorker
```

Comment:
A suggestion for this would be to rename it `FeedbackWorker` (or some other name) and add `Rpush.apns_feedback` call. This way it would help us get two giggles from one tickle (bird friendly alternative phrase):

- Keeping the current Redis logic will make sure we continue to add an expire time to all PNs that are continuously being added to Redis
   - Originally named `CleanupWorker` because that was the sole objective back when I spiked on this
- Keeping it as a scheduled job that runs every X minutes but adding `Rpush.apns_feedback` will help handle retries and other Reflection API callbacks periodically as well
   - I think it's every 8 minutes right now in `config/schedule.yml` (arbitrary number open to being tweaked as we see fit)

Reply:
On the other hand, wouldn't we lose the ability to load balance and and schedule these jobs separately? I'm worried that bundling together two separate functions (delivering notifications and clearing the older ones) might cause us problems when it comes to scaling these jobs!

Reply:
That's true! The redis logic shouldn't be much problem but I'd be totally okay if we have three separate background jobs, these two (redis cleanup can stay every 8h and the feedback one every ~10min?) and the main one that only triggers when new push notifications are require to be sent. 

This will definitely give us a better ability to tweak how often each runs 👍🏼